test/blocks/global-footer/global-footer.test.js

```diff
@@ -0,0 +1,263 @@
+import { expect } from '@esm-bundle/chai';
+import sinon, { stub } from 'sinon';
+import {
+  allElementsVisible,
+  alwaysVisibleSelectorsDesktop,
+  alwaysVisibleSelectorsMobile,
+  containerSelector,
+  createFullGlobalFooter,
+  insertDummyElementOnTop,
+  selectors,
+  waitForFooterToDecorate,
+} from './test-utilities.js';
+import baseFooter from './mocks/base-footer.js';
+import fetchedFooter from './mocks/fetched-footer.js';
+import icons from './mocks/icons.js';
+import { isElementVisible, mockRes } from '../global-navigation/test-utilities.js';
+import placeholders from './mocks/placeholders.js';
+
+const originalFetch = window.fetch;
+
+describe('global footer', () => {
+  let clock = null;
+  beforeEach(async () => {
+    document.body.innerHTML = baseFooter;
+    clock = sinon.useFakeTimers({
+      toFake: ['setTimeout'],
+      shouldAdvanceTime: true,
+    });
+
+    window.fetch = stub().callsFake((url) => {
+      if (url.includes('/footer')) {
+        return mockRes({
+          payload: fetchedFooter(
+            { regionPickerHash: '/fragments/regions#langnav' },
+          ),
+        });
+      }
+
+      if (url.includes('/placeholders')) return mockRes({ payload: placeholders });
+
+      if (url.includes('icons.svg')) return mockRes({ payload: icons });
+
+      return null;
+    });
+  });
+
+  afterEach(() => {
+    clock.restore();
+    window.fetch = originalFetch;
+  });
+
+  describe('wide screen', async () => {
+    it('should render the footer on wide screen', async () => {
+      const footer = await createFullGlobalFooter({ waitForDecoration: true, viewport: 'wide' });
+
+      expect(footer).to.exist;
```

Comment:
Since you already have some checks in the `createFullGlobalFooter` do you need the variable? The promise would reject and fail the test, potentially you can also just do:
```suggestion
      wait createFullGlobalFooter({ waitForDecoration: true, viewport: 'wide' });
```